src/librustdoc/html/render.rs

```diff
@@ -2292,20 +2292,35 @@ fn item_function(w: &mut fmt::Formatter, cx: &Context, it: &clean::Item,
                            f.generics).len();
     write!(w, "{}<pre class='rust fn'>", render_spotlight_traits(it)?)?;
     render_attributes(w, it)?;
-    write!(w, "{vis}{constness}{unsafety}{abi}fn \
-               {name}{generics}{decl}{where_clause}</pre>",
-           vis = VisSpace(&it.visibility),
-           constness = ConstnessSpace(f.constness),
-           unsafety = UnsafetySpace(f.unsafety),
-           abi = AbiSpace(f.abi),
-           name = it.name.as_ref().unwrap(),
-           generics = f.generics,
-           where_clause = WhereClause { gens: &f.generics, indent: 0, end_newline: true },
-           decl = Method {
-               decl: &f.decl,
-               name_len,
-               indent: 0,
-           })?;
+    if w.alternate() {
```

Comment:
What is this change for? `item_function` is only used to produce HTML.

Reply:
Because in the slidebar on the left, we call these function and escape the text once again. Guess what happens when you escape a string twice?

Reply:
This function isn't used for rendering the sidebar. This piece of code is rendering the `<pre>` block at the top of function pages.

Reply:
Yes but does it hurt to have this? Seems logical that the `alternate` handling is propagated to the children no? But if it's an issue, I'll just remove it.

Reply:
`alternate` is never set so this is an issue because it's dead code.

Reply:
Ok, I'll just remove the update in here then.